addons/website/static/src/scss/website.scss

```diff
@@ -207,20 +207,19 @@ body:not(.o_fullscreen) {
     .navbar-collapse {
         min-width: 0; // Allows it to shrink during loading
     }
-    .nav-item {
-        transition: opacity 1000ms ease 0s;
-    }
     .btn {
         // This was a default bootstrap style before but it was removed from
         // the library at some point. It seems important in the header so that
         // the header does not flicker during loading.
         white-space: nowrap;
     }
-    .o_menu_loading {
+    .o_menu_image_placeholder {
+        width: 80px !important;
+    }
+    #top_menu {
         flex-wrap: nowrap !important;
-        overflow: hidden !important;
-
-        .nav-item {
+        &.o_menu_loading {
+            overflow: hidden !important;
             opacity: 0 !important;
         }
     }
```

Comment:
Is this still needed at all? (even removing the class o_menu_loading from the XML would be possible I think? Since the menu is now updated before being shown anyway? Or is there maybe still a micro-flickering with F5-reloading or cache or whatever without it?)

Reply:
**I- About hidden overflow :**

I think the `overflow: hidden` CSS is very important since without it we won't be able to check if there is an overflow on the menu using JS (width calculation changes...).

**II- About opacity :**

This is a little bit confusing, I thought you mean by **"existing and invisible DOM"** the fact that `o_menu_loading` is applied on menu and this way **[A]**:

1- We start with an invisible menu (but DOM is available because of `opacity: 0`).
2- We do the first update (`_adapt()` method).
3- The menu is shown and we do the same thing once images are loaded and the flicker can only be caused by the difference: **[real width after loading images] - [width using default image width values] [B]**.

I think even on a non lazy loaded JS file we can't run the method in a moment when the whole DOM is loaded & invisible and that happens because the browser starts showing the DOM before it is completely loaded (something like FOUC).

**[A]** I get an acceptable result with this implementation:

https://user-images.githubusercontent.com/25032876/135154178-774d784e-5dea-484a-a36e-ee689ba1a3b2.mp4

**[B]** The flicker you got when page reloaded in your test [here](https://github.com/odoo/odoo/pull/75063#pullrequestreview-756470881) is caused by the lazy loaded `.o_lang_flag` image in language selector which has a non-resolved promise and the code (after all images are loaded) will never be executed this way...
I fixed that by excluding `.o_lang_flag` images from this process because I checked in [CSS and it seems that they already have a width value](https://github.com/odoo/odoo/blob/97ed683e3e7cc6b340bb78c4faa01a703a573f32/addons/website/static/src/scss/website.scss#L1300) so no need to take them into consideration :wink: